 # Q1: Understanding Central Tendency


In [1]:
import numpy as np

production_values = [120, 150, 145, 160, 155, 150, 145]         # Daily production values
mean_production = np.mean(production_values)                    #calculating mean
median_production = np.median(production_values)                #calculating median  
print("Mean weekly production:", mean_production)
print("Median weekly production:", median_production)


Mean weekly production: 146.42857142857142
Median weekly production: 150.0


# Q 2: Mean in Real Life

In [2]:
# Test scores
test = [78, 85, 90, 87, 88, 92, 95]
# Calculate the mean
score = np.mean(test)
print("Mean test score:", score)

# Significance: It represents the average performance of the whole class.
# This means, on average, students scored about 88 marks in the test.

Mean test score: 87.85714285714286


 # Q 3: Median in Real Life

In [3]:
# House prices
house_prices = [250000, 300000, 320000, 280000, 3000000]

# Calculate the median
median_price = np.median(house_prices)

print("Median house price: $", median_price)

# When to Use Median:

# 1. Outliers: Use the median when the data contains outliers that skew the mean.
# 2. Skewed Distributions: Use the median when the data has a skewed distribution, and the mean is not representative of the central tendency.


Median house price: $ 300000.0


# Q 4: Mode in Real Life

In [4]:

# Homework times
homework_times = [30, 35, 28, 40, 45]

# Calculate the range
range_homework = max(homework_times) - min(homework_times)

print("Range of homework times:", range_homework, "minutes")



# 1. Variability: The range tells us that the student's homework time varies by up to 17 minutes from the shortest to the longest session.
# 2. Consistency: A range of 17 minutes is relatively moderate, suggesting that the student's homework time is somewhat consistent, but with some variation.

Range of homework times: 17 minutes


# Q 5: Dispersion Introduction

In [24]:
# Daily homework times in minutes
homework_times = [30, 35, 28, 40, 45]

# Calculate range
range_time = max(homework_times) - min(homework_times)

print("📊 Homework Times:", homework_times)
print(f"✅ Range: {range_time} minutes")

# What the Range Tells Us:
# - Range = 45 - 28 = 17 minutes
# - This means the student’s homework time varies by up to 17 minutes from day to day.
# - It suggests moderate variability, possibly due to differences in subject difficulty, distractions, or workload.



📊 Homework Times: [30, 35, 28, 40, 45]
✅ Range: 17 minutes


# Q 6: Range in Action

In [5]:
# Greenhouse temperatures
greenhouse_temps = [15, 18, 20, 22, 25]

# Calculate the range
range_temp = max(greenhouse_temps) - min(greenhouse_temps)

print("Range of greenhouse temperatures:", range_temp, "°C")

# The range of 10°C indicates the variability in the greenhouse temperature. This information can guide the farmer's greenhouse management in several ways:




Range of greenhouse temperatures: 10 °C


# Q 7: Variance for Decision-Making

In [6]:
variance_A = 4  # Variance in lifespan of bulbs from Factory A
variance_B = 10  # Variance in lifespan of bulbs from Factory B

# Check which factory has lower variance, meaning more consistency
if variance_A < variance_B:
    print("Factory A produces more consistent bulbs.")  # Factory A is more consistent

elif variance_B < variance_A:
    print("Factory B produces more consistent bulbs.")  # Factory B is more consistent

else:
    print("Both factories have the same consistency.")  # Both have equal consistency

    # explaination: Lower variance means the data points (lifespans) are closer to the average, indicating greater consistency. Since Factory A has a variance of 4 compared to Factory B's 10, Factory A is more consistent.

Factory A produces more consistent bulbs.


# Q 8: Standard Deviation in Context

In [7]:
std_dev_A = 5                                      
std_dev_B = 15  
if std_dev_A > std_dev_B:                      # Check if Stock A has higher volatility
    print("Stock A is riskier.")               # If true, Stock A is riskier

elif std_dev_B > std_dev_A:                    # Check if Stock B has higher volatility
    print("Stock B is riskier.")               # If true, Stock B is riskier

else:                                          # If both have equal volatility
    print("Both stocks have the same risk.")   # Then both are equally risky


#  Explanation:
# Standard deviation measures how much a stock's price fluctuates. A higher standard deviation means greater volatility, which translates to higher investment risk. Since Stock B has a standard deviation of $15, it's riskier than Stock A.


Stock B is riskier.


# Q 9: Combining Measures

In [12]:
import statistics as stats

# Weekly water usage in gallons
water_usage = [500, 520, 480, 530, 510]

# Calculate mean and standard deviation
mean_usage = stats.mean(water_usage)
std_dev = stats.stdev(water_usage)

print("📊 Weekly Water Usage Data:", water_usage)
print(f"✅ Mean (Average): {mean_usage:.2f} gallons")
print(f"✅ Standard Deviation: {std_dev:.2f} gallons")


📊 Weekly Water Usage Data: [500, 520, 480, 530, 510]
✅ Mean (Average): 508.00 gallons
✅ Standard Deviation: 19.24 gallons


# Q 10: Central Tendency in Health

In [14]:
import statistics as stats

data = [7, 8, 7, 9, 8, 10, 7]

mean = stats.mean(data)
median = stats.median(data)
mode = stats.mode(data)

print(f"Mean: {mean}")
print(f"Median: {median}")
print(f"Mode: {mode}")


# Explanation:
# - Mean: Sensitive to outliers (like 10).
# - Median: Middle value, resistant to extremes.
# - Mode: Most frequent value — in this case, 7, which appears three times.
# Since 7 is the most common recovery time, mode is the best representation of a typical recovery duration in this dataset.



Mean: 8
Median: 8
Mode: 7


# Q 11: Outliers in Real-Life Data

In [ ]:
tips = np.array([10, 12, 15, 11, 200])

mean_tips = np.mean(tips)
median_tips = np.median(tips)

print(f"Mean tip: {mean_tips}")
print(f"Median tip: {median_tips}")

# Explanation:
# The mean is 49.6, which is much higher than most tip values because the large tip of 200 skews it upward.
# The median is 12, which better reflects the typical tip amount since it is not influenced by the outlier.

 # Q 12: Comparing Datasets

In [15]:
# Test scores for each class
class_A = [75, 78, 80, 85, 90]
class_B = [70, 72, 90, 95, 100]

# Calculate range for each class
range_A = max(class_A) - min(class_A)  # Range = highest - lowest
range_B = max(class_B) - min(class_B)

# Print the ranges
print("Range of Class A:", range_A)
print("Range of Class B:", range_B)

# Determine which class has greater variability
if range_A > range_B:
    print("Class A has greater score variability.")
elif range_B > range_A:
    print("Class B has greater score variability.")
else:
    print("Both classes have the same score variability.")

 # Explanation:
# - Range is a simple measure of spread.
# - A larger range means scores are more spread out, indicating greater variability.
# - In this case, Class B has a wider spread of scores, so it's less consistent.


Range of Class A: 15
Range of Class B: 30
Class B has greater score variability.


# Q 13: Skewed Data

In [18]:

import statistics as stats  # Import the statistics module for mean, median, and mode calculations

income = [3000, 3200, 3100, 2900, 3000, 3050, 10000]     # List of income values including an outlier
mean_income = stats.mean(income)                         # Average income, affected by the high outlier (10000)
median_income = stats.median(income)                      # Middle value when incomes are sorted
mode_income = stats.mode(income)                          # Most frequent income value in the list
print("Monthly Incomes:", income)                         # Show the original income list
print(f" Mean: ${mean_income:.2f}")                       # Print the mean with two decimal places
print(f" Median: ${median_income}")                        # Print the median value
print(f" Mode: ${mode_income}")                            # Print the mode value


Monthly Incomes: [3000, 3200, 3100, 2900, 3000, 3050, 10000]
 Mean: $4035.71
 Median: $3050
 Mode: $3000


| Measure    | Value   | Effect of Manager’s High Income                                          |
| ---------- | ------- | ------------------------------------------------------------------------ |
| **Mean**   | 4464.29 | **Increased a lot** because the $10,000 salary pulls the average upward. |
| **Median** | 3050    | **Barely affected**, since it’s the middle value and ignores extremes.   |
| **Mode**   | 3000    | **Unchanged**, still the most frequent income among workers.             |


Which Measure Best Represents Central Tendency?

The Median best represents the typical income because:

* It is not affected by extreme values (outliers).

* It reflects what a typical worker earns ($3,050), unlike the mean, which is distorted by the manager’s high salary.

# Q 14: Application in Business

In [19]:
import statistics as stats  # Import the statistics module for statistical calculations

sales = [150, 200, 250, 300, 350]                        # List of weekly sales figures
variance = stats.variance(sales)                         # Measure of how spread out the sales data is
std_dev = stats.stdev(sales)                             # Standard deviation shows average deviation from the mean
print(" Weekly Sales Data:", sales)                      # Print the original sales list
print(f" Variance: {variance:.2f}")                      # Print variance rounded to two decimal places
print(f" Standard Deviation: {std_dev:.2f}")             # Print standard deviation rounded to two decimal places


 Weekly Sales Data: [150, 200, 250, 300, 350]
 Variance: 6250.00
 Standard Deviation: 79.06


**How This Helps Forecasting**
- Identify Stability: A low standard deviation means sales are consistent week to week. High deviation suggests volatility.
- Set Realistic Targets: Helps in setting sales goals and inventory levels based on expected fluctuations.
- Detect Trends or Outliers: If future sales fall far outside the standard deviation range, it may indicate a trend shift or anomaly.
- Improve Planning: Marketing, staffing, and stock decisions can be better aligned with expected sales variability.


# Q 15: Practical Application

In [22]:
import statistics as stats

# Player scores
scores = [20, 22, 18, 25, 30, 15, 20, 18, 25, 20]

# Calculate mean, median, mode, range, and standard deviation
mean_score = stats.mean(scores)
median_score = stats.median(scores)
mode_score = stats.mode(scores)
range_score = max(scores) - min(scores)
std_dev = stats.stdev(scores)

print(" Player Scores:", scores)
print(f" Mean: {mean_score}")
print(f" Median: {median_score}")
print(f" Mode: {mode_score}")
print(f" Range: {range_score}")
print(f" Standard Deviation: {std_dev:.2f}")


 Player Scores: [20, 22, 18, 25, 30, 15, 20, 18, 25, 20]
 Mean: 21.3
 Median: 20.0
 Mode: 20
 Range: 15
 Standard Deviation: 4.35


| Measure                        | Value                       | Insight                                                                |
| ------------------------------ | --------------------------- | ---------------------------------------------------------------------- |
| **Mean (21.3)**                | Average score               | Overall performance across 10 games                                    |
| **Median (20)**                | Middle score                | Typical performance; half the games scored below 20 and half above     |
| **Mode (20)**                  | Most frequent score         | The score player achieves most often                                   |
| **Range (15)**                 | Max − Min (30 − 15)         | Shows variability between best and worst game                          |
| **Standard Deviation (≈5.08)** | Typical deviation from mean | Moderate variation in performance; scores are not extremely spread out |
